### PR TITLE
Refactor: Move `merge` from 'intrinsic_array_functions' to 'intrinsic_elemental_function'

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Towards: #4017

### CODE REVIEW

**Comment by Pranavchiku** [comment]
Brilliant, with this PR merged we won't need to workaround https://github.com/lfortran/lfortran/pull/4025#issuecomment-2118305042.

File: src/libasr/pass/intrinsic_functions.h [hunk 1]
```diff
@@ -2941,6 +2942,83 @@ namespace Maskr {
...
+namespace Merge {
+
+    static inline ASR::expr_t* eval_Merge(
+        Allocator &/*al*/, const Location &/*loc*/, ASR::ttype_t *,
+            Vec<ASR::expr_t*>& args, diag::Diagnostics& /*diag*/) {
+        LCOMPILERS_ASSERT(args.size() == 3);
+        ASR::expr_t *tsource = args[0], *fsource = args[1], *mask = args[2];
+        if( ASRUtils::is_array(ASRUtils::expr_type(mask)) ) {
+            return nullptr;
+        }
```
Comment (Pranavchiku) [inline src/libasr/pass/intrinsic_functions.h:2954]: This will never happen I think

File: src/libasr/pass/intrinsic_functions.h [hunk 2]
```diff
@@ -2941,6 +2942,83 @@ namespace Maskr {
...
+        return nullptr;
+    }
+
+    static inline ASR::expr_t* instantiate_Merge(Allocator &al,
+            const Location &loc, SymbolTable *scope,
+            Vec<ASR::ttype_t*>& arg_types, ASR::ttype_t *return_type,
+            Vec<ASR::call_arg_t>& new_args, int64_t /*overload_id*/) {
+        LCOMPILERS_ASSERT(arg_types.size() == 3);
+
+        // Array inputs should be elementalised in array_op pass already
```
Comment (Pranavchiku) [inline src/libasr/pass/intrinsic_functions.h:2972]: Yes, same argument in above comment.

**Review by Pranavchiku — APPROVED** [review]
This looks good, thanks for doing this. Just apply one comment above, rebase it and this PR will be good to merge.

File: src/libasr/pass/intrinsic_functions.h (hunk 2 above)
Comment (HarshitaKalani) [reply src/libasr/pass/intrinsic_functions.h:2972]: Oh right. Since now it is an elemental function.

**Pull request merged**